lib/aws-efs-creator-stack.ts

```diff
@@ -1,16 +1,65 @@
 import * as cdk from 'aws-cdk-lib';
 import { Construct } from 'constructs';
-// import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as efs from 'aws-cdk-lib/aws-efs';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { AwsEfsCreatorStackProps } from './AwsEfsCreatorStackProps';
 
 export class AwsEfsCreatorStack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+  constructor(scope: Construct, id: string, props: AwsEfsCreatorStackProps) {
     super(scope, id, props);
 
-    // The code that defines your stack goes here
+    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC`, {
+      vpcId: props.vpcId,
+    });
+    const removalPolicy = props.deployEnvironment === 'production' ? cdk.RemovalPolicy.RETAIN : cdk.RemovalPolicy.DESTROY;
 
-    // example resource
-    // const queue = new sqs.Queue(this, 'AwsEfsCreatorQueue', {
-    //   visibilityTimeout: cdk.Duration.seconds(300)
-    // });
+    // define a security group for EFS
+    const efsSG = new ec2.SecurityGroup(this, `${props.resourcePrefix}-efsSG`, {
+      securityGroupName: `${props.resourcePrefix}-efsSG`,
+      vpc: vpc,
+      allowAllOutbound: true,
+    });
+    efsSG.applyRemovalPolicy(removalPolicy);
+
+    // create an EFS File System
+    const efsFileSystem = new efs.FileSystem(this, `${props.resourcePrefix}-efsFileSystem`, {
+      fileSystemName: `${props.resourcePrefix}-efsFileSystem`,
+      vpc,
+      removalPolicy,
+      securityGroup: efsSG, // Ensure this security group allows NFS traffic from the ECS tasks
+      encrypted: true, // Enable encryption at rest
+      performanceMode: efs.PerformanceMode.GENERAL_PURPOSE, // For AI application, HCP application, Analytics application, and media processing workflows
+      allowAnonymousAccess: false, // Disable anonymous access
+      throughputMode: efs.ThroughputMode.ELASTIC,
+      lifecyclePolicy: efs.LifecyclePolicy.AFTER_30_DAYS, // After 2 weeks, if a file is not accessed for given days, it will move to EFS Infrequent Access.
+    });
+
+    // add EFS access policy
+    efsFileSystem.addToResourcePolicy(
+        new iam.PolicyStatement({
+            actions: ['elasticfilesystem:ClientMount'],
+            principals: [new iam.AnyPrincipal()],
+            conditions: {
+                Bool: {
+                    'elasticfilesystem:AccessedViaMountTarget': 'true'
+                }
+            },
+        }),
```

Comment:
**Suggestion:** The `efsFileSystem` resource policy allows any principal to mount the file system, which can be a security risk. Consider restricting the principals to a specific role or set of roles.
> 🌖  **High Priority** 
 > [category: security, importance: 9]
```suggestion
    efsFileSystem.addToResourcePolicy(
        new iam.PolicyStatement({
            actions: ['elasticfilesystem:ClientMount'],
            principals: [new iam.ArnPrincipal('arn:aws:iam::123456789012:role/SpecificRole')],
            conditions: {
                Bool: {
                    'elasticfilesystem:AccessedViaMountTarget': 'true'
                }
            },
        }),
    );
```